### ISSUE TITLE
[Feature] Support alter table to update properties and custom properties

### ISSUE THREAD

**Issue by swuferhong** [alibaba/fluss#100]
### Search before asking

- [X] I searched in the [issues](https://github.com/alibaba/fluss/issues) and found nothing similar.

### Motivation

Support alter table to update properties and custom properties.

### Willingness to contribute

- [X] I'm willing to submit a PR!

**Comment by wuchong** [comment]
@gyang94 assigned to you.

**Comment by leonardBang** [comment]
kindly ping @gyang94, we planed to code freeze soon, how about this feature development?

**Comment by gyang94** [comment]
@leonardBang haven't started. This was assigned to me just 4 days ago.
will this feature be released in this version?

**Comment by leonardBang** [comment]
No,  please keep your pace @gyang94 , I just want to check all subtask status

**Comment by leonardBang** [comment]
This is not a blocker for 0.7 version

**Comment by LiebingYu** [comment]
Hi @gyang94. Kindly ask wha's the progress about this? #327 may depend on this.

**Comment by LiebingYu** [comment]
Changing the number of table buckets(#327) is an important feature that depends on this functionality and needs to be supported as soon as possible. Since @gyang94 has not responded for a long time. I think I can take over this task. CC @wuchong @swuferhong.

**Comment by gyang94** [comment]
@LB-Yu Sorry for late response. I didn't receive the notification about your comment last week, and I am not aware that this is a blocker for other ticket.
Recently I am working on iceberg bucketing function part, and just finished coding in last weekend. Now I can move to this ticket. I will finish this ticket in this week, is that good for you?
